### DISCUSSION TITLE
bagaimana memunculkan Interactive Map

### DISCUSSION THREAD

**Discussion by irwan547** [teamalgoritma/community#561] [Q&A] [answered]
selamat pagi,

mohon pencerahannya saya salah dimana ya dalam coding untuk memunculkan interactive map?

![image](https://user-images.githubusercontent.com/49586368/220509311-de13818f-fdbb-42c8-9359-03755519efc2.png)

![image](https://user-images.githubusercontent.com/49586368/220507208-52813269-fb21-460f-a504-ceb211ab05e6.png)

saya sudah mencoba rumusan lain namun tidak bisa juga
![image](https://user-images.githubusercontent.com/49586368/220512291-4edf4eb7-8e23-498e-ae7a-09284056f740.png)

namun ketika saya masukkan koordinat secara manual, codingnya berhasil
![image](https://user-images.githubusercontent.com/49586368/220507731-9d424a45-968d-475b-95bd-e54cc978c35c.png)

apakah bisa dibantu ?

Terimakasih

**Comment by finesaaa** [comment, marked as answer]
Dear pak @irwan547,
Terima kasih atas pertanyaannya, akan dijawab segera oleh tim kami ya pak

**Reply by finesaaa** [reply]
![image](https://user-images.githubusercontent.com/57583780/220519959-1eb9f7c5-5cdf-4959-b101-fbeac605960b.png)
Ada kesalahan pada subsetting di lat, lon - harusnya seperti ini pak
```
lat = earthquakes2[earthquakes2$remark %in% "Arafura Sea", ]$lat,
lng = earthquakes2[earthquakes2$remark %in% "Arafura Sea", ]$lon
```

**Reply by irwan547** [reply]
oh iya terimakasih mbak @finesaaa , betul mbak, sekarang sudah bisa dijalankan :)

saya ada 2 pertanyaan lagi,

- untuk mengubah icon 
![image](https://user-images.githubusercontent.com/49586368/220520965-0d62173b-f684-4507-ad02-6316c032233e.png)
menjadi icon berupa titik ( **.** ) caranya bagaimana? karena jika titiknya banyak, mapnya akan padat dengan icon tersebut

- dan yang saya alami saat ini adalah dengan adanya >90,000 row latitude dan longitude, komputernya langsung ngehang dan tidak bisa menampilkan posisi di map.  bahkan saat saya subset untuk suatu region dengan >5,000 row juga komputernya ngehang. apakah ada cara untuk membuat processingnya menjadi ringan dan cepat?

**Reply by finesaaa** [reply]
Dear @irwan547 ,

1. Terkait kustomisasi icon silahkan dapat diekplorasi sesuai dokumentasi https://leafletjs.com/examples/custom-icons/, nanti bapak bisa custom pakai image (.) dan sebagainya.
2. Kemudian terkait banyaknya data yang membuat hang, memang untuk leaflet memiliki keterbatasan load data, berdasarkan diskusi pada StackOverflow berikut.
https://stackoverflow.com/questions/64082365/r-leaflet-limitations-too-much-data
Namun, disini bapak mungkin dapat mencoba leafgl - https://github.com/r-spatial/leafgl, yang dikatakan dapat merender leaflet dengan >1M points
